tmva/tmva/src/MethodLD.cxx

```diff
@@ -174,7 +174,8 @@ Double_t TMVA::MethodLD::GetMvaValue( Double_t* err, Double_t* errUpper )
       (*fRegressionReturnVal)[iout] = (*(*fLDCoeff)[iout])[0] ;
 
       int icoeff=0;
-      for (std::vector<Float_t>::const_iterator it = ev->GetValues().begin();it!=ev->GetValues().end();++it){
+      std::vector<Float_t> values = ev->GetValues();
+      for (std::vector<Float_t>::const_iterator it = values.begin();it!=values.end();++it) {
```

Comment:
```suggestion
      for (Float_t val : ev->GetValues()){
```
The assertion failure seems to be a false positive, because `Event::GetValues()` actually returns a reference.

I agree it's not nice that `GetValues()` is called twice in this iterator loop, but right now you are doing one additional copy. Why not use a range-based loop?